examples/name-service-without-macros/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::bytes::AsByteVec;
+use light_sdk::account_info::LightAccountInfo;
+use light_sdk::{
+    compressed_account::{LightAccount, LightAccounts},
+    instruction_data::LightInstructionData,
+    light_system_accounts,
+    verify::verify_compressed_accounts,
+    LightDiscriminator, LightHasher, LightTraits,
+};
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+    use light_sdk::account_info::convert_metas_to_infos;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateRecord<'info>>,
+        inputs: Vec<u8>,
+        name: String,
+        rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+        let mut light_accounts = LightCreateRecord::try_light_accounts(&account_infos)?;
+
+        light_accounts.record.derive_address(
+            &[b"name-service", name.as_bytes()],
+            &crate::ID,
+            ctx.remaining_accounts,
+        );
+
+        light_accounts.record.owner = ctx.accounts.signer.key();
+        light_accounts.record.name = name;
+        light_accounts.record.rdata = rdata;
+
+        verify_compressed_accounts(
+            &ctx,
+            inputs.proof,
+            &[light_accounts.record],
+            None,
+            false,
+            None,
+            &crate::ID,
+        )?;
+
+        Ok(())
+    }
+
+    pub fn update_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, UpdateRecord<'info>>,
+        inputs: Vec<u8>,
+        new_rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
```

Comment:
Hmm... I would prefer to take `[&u8]` as an argument, I don't want to burden developers with having to construct a cursor manually.

Reply:
to be honest imo this function is too highlevel. In particular, once we add more specialized `LightAccount` and `LightAccountInfo` variations this struct will not work anymore.
Also for macros we shouldn't use this since it adds additional overhead for future maintenance when we extend the sdk.